system/jlib/jutil.cpp

```diff
@@ -2873,24 +2873,84 @@ jlib_decl StringBuffer &getTempFilePath(StringBuffer & target, const char * comp
     StringBuffer dir;
     if (pTree)
         getConfigurationDirectory(pTree->queryPropTree("Directories"),"temp",component,pTree->queryProp("@name"),dir);
-    if (!dir.length())
+    bool ok = false;
+    if (!dir.isEmpty())
     {
+        ok = recursiveCreateDirectory(dir.str());
+        if (ok)
+        {
 #ifdef _WIN32
-        char path[_MAX_PATH+1];
-        if(GetTempPath(sizeof(path),path))
-            dir.append(path).append("HPCCSystems\\hpcc-data");
-        else
-            dir.append("c:\\HPCCSystems\\hpcc-data\\temp");
+            return target.set(dir);
+#else
+            int srtn = access(dir.str(), R_OK|W_OK|X_OK);
+            if (!srtn)
+                return target.set(dir);
+#endif
+        }
+    }
+
+    // runtime dir
+    dir.clear();
+    dir.append(RUNTIME_DIR);
+    dir.append(PATHSEPCHAR).append("hpcc-data").append(PATHSEPCHAR).append(component);
+    dir.append(PATHSEPCHAR).append("temp");
+    ok = recursiveCreateDirectory(dir.str());
+    if (ok)
+    {
+#ifdef _WIN32
+        return target.set(dir);
```

Comment:
Why no access check on Windows?

Reply:
From MSDN documentation: "in Windows 2000 and later operating systems, all directories have read and write access" - so I didn't feel a specific check was required.
To check directory permissions accurately may require much more Windows specific Access Token code.

Reply:
I can add a call to access(R_OK|W_OK) for Windows to be more similar.